coalib/coala_main.py

```diff
@@ -60,15 +62,23 @@ def run_coala(console_printer=None,
     :param arg_parser:              Instance of ArgParser that is used to parse
                                     non-setting arguments.
     :param arg_list:                The CLI argument list.
+    :param args:                    Alternative pre-parsed CLI arguments.
+    :param debug:                   Run in debug mode, bypassing
+                                    multiprocessing, and not catching any
+                                    exceptions.
     :return:                        A dictionary containing a list of results
                                     for all analyzed sections as key.
     """
+    assert not (arg_list and args), (
+        'Either provide run_coala() with an arg_list of CLI arguments or '
+        'pre-parsed args, but not both.')
```

Comment:
can you put this check into gather_configuration instead? This only gets passed through here